tools/indexer/wallet/src/main.rs

```diff
@@ -0,0 +1,221 @@
+mod schema;
+use schema::access_keys;
+
+const ADD: &str = "ADD";
+const DELETE: &str = "DELETE";
+const PENDING: &str = "PENDING";
+const SUCCESS: &str = "SUCCESS";
+const FAILED: &str = "FAILED";
+const FUNCTION_CALL: &str = "FUNCTION_CALL";
+const FULL_ACCESS: &str = "FULL_ACCESS";
+
+#[derive(Insertable, Queryable)]
+struct AccessKey {
+    public_key: String,
+    account_id: String,
+    action: &'static str,
+    status: &'static str,
+    receipt_hash: String,
+    block_height: BigDecimal,
+    permission: &'static str,
+}
+
+impl AccessKey {
+    pub fn from_receipt_view(
+        receipt: &near_indexer::near_primitives::views::ReceiptView,
+        block_height: u64,
+    ) -> Vec<Self> {
+        let mut access_keys: Vec<Self> = vec![];
+        if let near_indexer::near_primitives::views::ReceiptEnumView::Action { actions, .. } =
+            &receipt.receipt
+        {
+            for action in actions {
+                let access_key = match action {
+                    near_indexer::near_primitives::views::ActionView::AddKey {
+                        public_key,
+                        access_key,
+                    } => Self {
+                        public_key: public_key.to_string(),
+                        account_id: receipt.receiver_id.to_string(),
+                        action: ADD,
+                        status: PENDING,
+                        receipt_hash: receipt.receipt_id.to_string(),
+                        block_height: BigDecimal::from_u64(block_height).expect("Block height must be provided."),
+                        permission: match access_key.permission {
+                            near_indexer::near_primitives::views::AccessKeyPermissionView::FullAccess => {
+                                FULL_ACCESS
+                            },
+                            near_indexer::near_primitives::views::AccessKeyPermissionView::FunctionCall { .. } => {
+                                FUNCTION_CALL
+                            },
+                        },
+                    },
+                    near_indexer::near_primitives::views::ActionView::DeleteKey { public_key } => Self {
+                        public_key: public_key.to_string(),
+                        account_id: receipt.receiver_id.to_string(),
+                        action: DELETE,
+                        status: PENDING,
+                        receipt_hash: receipt.receipt_id.to_string(),
+                        block_height: BigDecimal::from_u64(block_height).expect("Block height must be provided."),
+                        permission: "",
+                    },
+                    _ => { continue },
+                };
+                access_keys.push(access_key);
+            }
+        }
+        access_keys
+    }
+}
+
+fn establish_connection() -> Pool<ConnectionManager<PgConnection>> {
+    dotenv().ok();
+
+    let database_url = env::var("DATABASE_URL").expect("DATABASE_URL must be set");
```

Comment:
Let's call it `NEAR_INDEXER_FOR_WALLET_DATABASE_URL`, otherwise, we may collide with something completely different.